hack/ocget/ocget.go

```diff
@@ -0,0 +1,185 @@
+		}
+	}
+	return false
+}
+
+// isDouble indicates if we should ignore a given GroupKind because it is
+// accessible via a different API route.
+func isDouble(gk schema.GroupKind) bool {
+	switch gk.String() {
+	case "ClusterRole.authorization.openshift.io", // ClusterRole.rbac.authorization.k8s.io
+		"ClusterRoleBinding.authorization.openshift.io", // ClusterRoleBinding.rbac.authorization.k8s.io
+		"Role.authorization.openshift.io",               // Role.rbac.authorization.k8s.io
+		"RoleBinding.authorization.openshift.io",        // RoleBinding.rbac.authorization.k8s.io
+		"DaemonSet.extensions",                          // DaemonSet.apps
+		"Deployment.extensions",                         // Deployment.apps
+		"ImageStreamTag.image.openshift.io",             // ImageStream.image.openshift.io
+		"ReplicaSet.extensions",                         // ReplicaSet.apps
+		"Project.project.openshift.io",                  // Namespace
+		"SecurityContextConstraints":                    // SecurityContextConstraints.security.openshift.io
+		return true
+	}
+
+	return false
+}
+
+func (g *getter) GetOrList(ctx context.Context, kind, namespace, name string) error {
+	fmt.Printf("Get(%s, %s, %s)\n", kind, namespace, name)
+
+	if strings.EqualFold("Secret", kind) {
+		return nil
+	}
+
+	isList := name == "" || namespace == ""
+	for _, gr := range g.grs {
+		for version, resources := range gr.VersionedResources {
+			for _, resource := range resources {
+				if strings.ContainsRune(resource.Name, '/') { // no subresources
+					continue
+				}
+				if isList && !contains(resource.Verbs, "list") {
+					continue
+				}
+				if !isList && !contains(resource.Verbs, "get") {
+					continue
+				}
+
+				gvk := schema.GroupVersionKind{Group: gr.Group.Name, Version: version, Kind: resource.Kind}
+				gk := gvk.GroupKind()
+				if isDouble(gk) {
+					continue
+				}
+				if !strings.EqualFold(gk.String(), kind) {
+					continue
+				}
+				dc, err := g.dyn.ClientForGroupVersionKind(gvk)
+				if err != nil {
+					return err
+				}
+
+				fmt.Printf(" =%v \n", gvk.String())
+				if isList {
+					o, err := dc.Resource(&resource, namespace).List(metav1.ListOptions{})
+					if err != nil {
+						return err
+					}
+
+					l, ok := o.(*unstructured.UnstructuredList)
+					if !ok {
+						return nil
+					}
+					for _, i := range l.Items {
+						if (name != "" && i.GetName() == name) || name == "" {
+							fmt.Printf("  %s/%v\n", i.GetNamespace(), i.GetName())
+						}
+					}
+				} else {
+					o, err := dc.Resource(&resource, namespace).Get(name, metav1.GetOptions{})
+					if err != nil {
+						return err
+					}
+
+					fmt.Printf(" %s/%v\n", o.GetNamespace(), o.GetName())
+				}
+				return nil
+			}
+		}
+	}
+	return nil
+}
+
+func main() {
+	ctx := context.Background()
+	log := logrus.NewEntry(logrus.StandardLogger())
+	logrus.SetLevel(logrus.DebugLevel)
+
+	g, err := new(log)
+	if err != nil {
+		panic(err)
+	}
+	err = g.GetOrList(ctx, "Secret", "", "")
```

Comment:
There is a pattern here... should this be refactored into a loop with the different `GetOrList` parameters?

Reply:
it's just a test program, it will totally go away. this functionality will be behind the admin api for geneva actions to execute